caip-300.md

```diff
@@ -0,0 +1,223 @@
+  - params - will include params specific to the RPC method
+
+The methods that are allowed to be batched together in the `wallet_connect` request is going to be going to dependant on each ecosystem that must be defined under the [Namespaces][namespaces] repo to define a CAIP-300 profile
+
+### Response
+
+The wallet will prompt the user with a dedicated UI to display the app requesting the authentication and allow the user to select which blockchain account to sign with.
+
+#### Success
+
+If approved, the wallet will return a list of signed, valid CACAOs for each account authorized on the networks requested by the `chains` property.
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": { "result": unknown } || { error: { message: string, code: number } }[]
+}
+```
+
+The JSON-RPC response will include an array of requests which are going to be ordered in the same order as provided in the request.
+
+#### Failure
+
+Request will fail if rejected by the user or if parameters fail validation.
+
+The following Error responses MUST be used:
+
+- User Rejected Request
+  - code = 7000
+  - message = "User Rejected Connect"
+- Invalid Request Params
+  - code = 7001
+  - message = "Invalid Method Requested"
+
+### Fallback Behavior
+
+For backwards-compatibility, there is going to be a fallback behavior that is expected but is going to dependant on each ecosystem that must be defined under the [Namespaces][namespaces] repo to define a CAIP-300 profile.
+
+If we take for example Ethereum this would fallback to the legacy behavior of `eth_requestAccounts` as defined by ERC-1102
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "wallet_connect",
+  "params": {} // fallback to ERC-1102 request
+}
+```
+
+Which then will result in a single item in the array that will match the response of the fallback method defined for each Namespace profile for CAIP-300.
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": [
+    {
+      "result": ["0xa...", "0xb..."] // fallback to ERC-1102 request
+    }
+  ]
+}
+```
+
+## Rationale
+
+An RPC method was choosen for batching because it enabled a smoother upgrade path for existing dapps and wallets to support it without much breaking changes. While the alternative would be to batch RPC requests at the transport layer this is not possible today without a breaking change.
+
+Therefore batching inside a single RPC request makes it easier to re-use the existing developer tooling and will enable both dapps and wallet to progressively support this new "connection" standard for wallets.
+
+The rationale for making this as generic as possible is to enable profiles to be defined for different ecosystems such as Bitcoin, Ethereum, Solana, Cosmos, Polkadot, etc. The alternative would be to create specific fields for "session", "authentication", "capabilities", "permissions", etc. But this would introduce a bias from one chain to another and not favor chain-agnosticisim.
```

Comment:
```suggestion
The rationale for making this as generic as possible is to enable profiles to be defined for different ecosystems such as Bitcoin, Ethereum, Solana, Cosmos, Polkadot, etc. The alternative would be to create specific fields for "session", "authentication", "capabilities", "permissions", etc. But this would introduce a bias from one chain to another and harm chain-agnosticism.
```